Doc/tutorial/classes.rst

```diff
@@ -475,12 +475,20 @@ Random Remarks
 
 .. These should perhaps be placed more carefully...
 
-Data attributes override method attributes with the same name; to avoid
-accidental name conflicts, which may cause hard-to-find bugs in large programs,
-it is wise to use some kind of convention that minimizes the chance of
-conflicts.  Possible conventions include capitalizing method names, prefixing
-data attribute names with a small unique string (perhaps just an underscore), or
-using verbs for methods and nouns for data attributes.
+If the same attribute name occurs in both an instance and in a class,
+then attribute lookup prioritizes the instance::
```

Comment:
Addendum: The exception is with implicit lookup of (reserved) dunder names.
```
class C:
    def __len__(self): return 0

c = C()
c.__len__ = False
print(c.__len__, len(c))
# False, 0
```
Do you want to mention this, or qualify with 'direct attribute lookup'?  This is something that occasionally trips up newbies when they, for instance, add (in this instance) a .__len__ function to an instance and find that indirect access via len() still gets the class method.

Reply:
I think that level of detail is more suited for the language reference.  In a tutorial, it just invites confusion.  Special handling of dunder methods, data descriptors, and non-data descriptors is an advanced topic.

Reply:
Agreed.